lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -120,8 +125,17 @@ void HWMemSimImplPass::generateMemory(HWModuleOp op, FirMemory mem) {
     Value en = op.body().getArgument(inArg++);
     Value clock = op.body().getArgument(inArg++);
     // Add pipeline stages
-    en = addPipelineStages(b, mem.readLatency, clock, en);
-    addr = addPipelineStages(b, mem.readLatency, clock, addr);
+    if (ignoreReadEnableMem) {
+      for (size_t j = 0, e = mem.readLatency; j != e; ++j) {
+        auto enLast = en;
+        if (j < e - 1)
+          en = addPipelineStages(b, 1, clock, en);
```

Comment:
This differs from the SFC lowering in that it only generates a read enable if it is actually needed for the next cycle. The SFC generates one always even if it is never used, i.e., this is the last read enable.